lighthouse/helpers/plates.py

```diff
@@ -215,8 +221,68 @@ def row_to_dict(row):
     return obj
 
 
+def request_with_retries(request_func: Callable, response_func: Callable[[Any], T], max_retries=3) -> T:
+    attempt = 1
+
+    while attempt <= max_retries:
+        LOGGER.debug(f"Starting attempt {attempt} of {max_retries}...")
+
+        try:
+            response = request_func()
+
+            if response.status_code == 200:
+                return response_func(response)
+
+            LOGGER.debug(
+                f"Attempt failed due to an invalid status code {response.status_code}. "
+                "Pausing 1 second before trying again."
+            )
+            attempt += 1
+            sleep(1)
+        except requests.ConnectionError:
+            LOGGER.debug("Attempt failed due to a connection error. Pausing 1 second before trying again.")
+            attempt += 1
+            sleep(1)
+
+    raise requests.ConnectionError("Unable to access Sequencescape.")
+
+
+def plate_exists_in_ss_with_uuid(uuid: str) -> bool:
+    ss_url = f"{app.config['SS_URL']}/api/v2/plates"
+    params = {"filter[uuid]": uuid}
+    headers = _ss_headers()
+
+    LOGGER.debug(f"Searching Sequencescape for plate with UUID: '{uuid}'.")
+
+    return request_with_retries(
+        request_func=lambda: requests.get(ss_url, params=params, headers=headers),
+        response_func=lambda response: len(response.json()["data"]) > 0,
+    )
+
+
+def filter_for_new_samples(samples: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    def is_sample_new(sample: Dict[str, str]) -> bool:
```

Comment:
Fair enough.  We do have a lot of other instances of doing it this way and it does help the reader to identify the scope of the usage for this function, but I also understand why you might not like it.  I had always considered this approach as the method being part of the one it exists inside, hence it would be tested by the behaviour of the outer method.